### PR TITLE
Manager to run mount propagation test instead of the driver-deployer

### PR DESCRIPTION
**Pull request by JacieChao**
Fixed issue(https://github.com/rancher/longhorn/issues/129)

### CODE REVIEW

File: controller/node_controller.go [hunk 1]
```diff
@@ -354,3 +361,35 @@ func (nc *NodeController) syncDiskStatus(node *longhorn.Node) error {
...
+	for _, mount := range pod.Spec.Containers[0].VolumeMounts {
+		if mount.Name == types.LonghornSystemKey {
+			mountPropagationStr := ""
+			if mount.MountPropagation == nil {
+				mountPropagationStr = "nil"
+			} else {
+				mountPropagationStr = string(*mount.MountPropagation)
+			}
+			if mount.MountPropagation == nil || *mount.MountPropagation != v1.MountPropagationBidirectional {
+				logrus.Debugf("The MountPropagation value %s is not expected from pod %s, node %s", mountPropagationStr, pod.ObjectMeta.Name, pod.Spec.NodeName)
```
Comment (yasker) [inline controller/node_controller.go]: This log will continuing print if mount propagation wasn't enabled? Sounds like a bad idea to flood the log. Let's print the log only if the newly detected status is different from the value stored in the `node.Status.MountPropagation` (which covers both value is true and false).

**Comment by yasker** [comment]
Have you tested both cases?  (mount propagation enabled for cluster and disabled).

**Comment by JacieChao** [comment]
@yasker Yeah, I have tested enable/disabled mount propagation cases.

**Comment by yasker** [comment]
Wait, the order of commit is wrong. I don't think `driver: Update MountPropagation option detect function` can compile without the updated client.

**Comment by JacieChao** [comment]
@yasker Sorry about that. After separate the commit, the date of commit is different which caused a wrong order... I have fixed it.

**Comment by yasker** [comment]
Have you tested the final commit? The updated of the go client breaks the CSI plugin.

```
[longhorn-csi-plugin-drjnl longhorn-csi-plugin] time="2018-08-01T16:35:51Z" level=info msg="ControllerServer ControllerPublishVolume req: volume_id:"pvc-008b9885-95a7-11e8-8a76-def8917872c5" node_id:"yasker-longhorn-dev-1" volume_capability:<mount:<fs_type:"ext4" > access_mode:<mode:SINGLE_NODE_WRITER > > volume_attributes:<key:"storage.kubernetes.io/csiProvisionerIdentity" value:"1533140321748-8081-rancher.io/longhorn" > "
[longhorn-csi-plugin-drjnl longhorn-csi-plugin] E0801 16:35:51.484006       1 utils.go:100] GRPC error: rpc error: code = Internal desc = Failed to parse: {"actions":{"attach":"http://longhorn-backend:9500/v1/volumes/pvc-008b9885-95a7-11e8-8a76-def8917872c5?action=attach","engineUpgrade":"http://longhorn-backend:9500/v1/volumes/pvc-008b9885-95a7-11e8-8a76-def8917872c5?action=engineUpgrade","recurringUpdate":"http://longhorn-backend:9500/v1/volumes/pvc-008b9885-95a7-11e8-8a76-def8917872c5?action=recurringUpdate","replicaRemove":"http://longhorn-backend:9500/v1/volumes/pvc-008b9885-95a7-11e8-8a76-def8917872c5?action=replicaRemove"},"conditions":{"scheduled":{"lastProbeTime":"2018-08-01T16:22:49Z","lastTransitionTime":"2018-08-01T16:22:49Z","message":"","reason":"","status":"True","type":"scheduled"}},"controllers":[{"address":"","currentImage":"","endpoint":"","engineImage":"rancher/longhorn-engine:31c42f0","hostId":"","name":"pvc-008b9885-95a7-11e8-8a76-def8917872c5-e-934ad68a","running":false}],"created":"2018-08-01 16:22:06 +0000 UTC","currentImage":"rancher/longhorn-engine:31c42f0","engineImage":"rancher/longhorn-engine:31c42f0","fromBackup":"","frontend":"blockdev","id":"pvc-008b9885-95a7-11e8-8a76-def8917872c5","links":{"self":"http://longhorn-backend:9500/v1/volumes/pvc-008b9885-95a7-11e8-8a76-def8917872c5"},"migrationNodeID":"","name":"pvc-008b9885-95a7-11e8-8a76-def8917872c5","numberOfReplicas":2,"recurringJobs":null,"replicas":[{"address":"","currentImage":"","dataPath":"/var/lib/rancher/longhorn/replicas/pvc-008b9885-95a7-11e8-8a76-def8917872c5-80ffafe0","diskID":"681de48e5fe8bbb9","engineImage":"rancher/longhorn-engine:31c42f0","failedAt":"","hostId":"yasker-longhorn-dev-3","mode":"","name":"pvc-008b9885-95a7-11e8-8a76-def8917872c5-r-bcc4d853","running":false},{"address":"","currentImage":"","dataPath":"/var/lib/rancher/longhorn/replicas/pvc-008b9885-95a7-11e8-8a76-def8917872c5-953d8e44","diskID":"681de48e5fe8bbb9","engineImage":"rancher/longhorn-engine:31c42f0","failedAt":"","hostId":"yasker-longhorn-dev-1","mode":"","name":"pvc-008b9885-95a7-11e8-8a76-def8917872c5-r-c0f42c6b","running":false}],"robustness":"","size":"2147483648","staleReplicaTimeout":20,"state":"detached","type":"volume"}
[longhorn-csi-plugin-drjnl longhorn-csi-plugin] : json: cannot unmarshal object into Go struct field Volume.controllers of type string
```

**Comment by yasker** [comment]
I've fixed it at https://github.com/rancher/longhorn-manager/commit/1ed970ca96426082ad7b4093e8853be674ba3362

**Pull request merged**